airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/RemainingCredits.tsx

```diff
@@ -25,17 +39,98 @@ const StarIcon = styled(FontAwesomeIcon)`
   margin-right: 6px;
   font-size: 22px;
 `;
+const Actions = styled.div`
+  display: flex;
+  gap: 12px;
+`;
+
+const STRIPE_SUCCESS_QUERY = "stripeCheckoutSuccess";
+
+/**
+ * Checks whether the given cloud workspace had a recent increase in credits.
+ */
+function hasRecentCreditIncrease(cloudWorkspace: CloudWorkspace): boolean {
+  const lastIncrement = cloudWorkspace.lastCreditPurchaseIncrementTimestamp;
+  return lastIncrement ? Date.now() - lastIncrement < 30000 : false;
+}
 
 const RemainingCredits: React.FC = () => {
+  const retryIntervalId = useRef<number>();
+  const config = useConfig();
   const currentWorkspace = useCurrentWorkspace();
   const cloudWorkspace = useGetCloudWorkspace(currentWorkspace.workspaceId);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const invalidateWorkspace = useInvalidateCloudWorkspace(
+    currentWorkspace.workspaceId
+  );
+  const { isLoading, mutate: createCheckout } = useStripeCheckout((session) => {
```

Comment:
Maybe it makes more sense to use `mutateAsync` and then `const session = await createCheckout()`?

Also `mutate` and `mutateAsync` has `onSuccess` properties as second argument, so it may make more sense to pass it there directly and do not use it in hook.

Reply:
That would indeed look way better. Sorry I've missed that this function exists in the react-query docs, thanks for pointnig it out.